src/box/lua/misc.cc

```diff
@@ -197,6 +197,68 @@ lua_wrap_formatted_array(struct lua_State *L, struct tuple_format *format)
 	lua_setmetatable(L, -2);
 }
 
+/**
+ * Advances the iterator - invokes iterator_next() with a saved state.
+ * For details, see description of port_c_push_iterator_lua().
+ */
+static int
+port_c_iterator_lua_next(lua_State *L)
+{
+	struct port_c_iterator *iter =
+		(struct port_c_iterator *)lua_touserdata(L, 1);
+	bool is_eof = false;
+	struct port port;
+	int rc = iter->next(iter, &port, &is_eof);
+	if (rc != 0)
+		luaT_error(L);
+	if (is_eof)
+		return 0;
+	int top_svp = lua_gettop(L);
+	port_dump_lua(&port, L, PORT_DUMP_LUA_MODE_FLAT);
+	port_destroy(&port);
+	return lua_gettop(L) - top_svp;
+}
+
+/**
+ * The function should be called as a closure with one upvalue:
+ * port_c_iterator stored as userdata.
+ * For details, see description of port_c_push_iterator_lua().
+ */
+static int
+port_c_iterator_lua_start(lua_State *L)
```

Comment:
Nit: Let's call `port_c_iterator_start_lua` because we use `_lua` as a suffix in all other function names in this file.

Reply:
Changed the names a bit in this place:
1. All the functions are called `port_c_iterator_{action}_lua`.
2. But metatable and name are called `port_c_iterator_lua_{name,meta}` - it seems to be that `port_c_iterator_meta_lua` sounds weird.